### ISSUE TITLE
manhole_ssh emits warnings on import

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2912]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#2912|
|Type|defect|
|Created|2007-11-19 17:33:08Z|
|Branch|https://github.com/twisted/twisted/tree/manhole-ssh-warning-2912|

This was recently introduced to trunk:
```
Python 2.5.1 (r251:54863, Oct  5 2007, 13:36:32) 
[GCC 4.1.3 20070929 (prerelease) (Ubuntu 4.1.2-16ubuntu2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from twisted.conch import manhole_ssh
/home/glyph/Projects/Twisted/trunk/twisted/conch/manhole_ssh.py:125: DeprecationWarning: getPublicKeyString is deprecated since Twisted Conch 0.9.  Use Key.fromString().
  'ssh-rsa' : keys.getPublicKeyString(data = publicKey)
/home/glyph/Projects/Twisted/trunk/twisted/conch/manhole_ssh.py:142: DeprecationWarning: getPrivateKeyObject is deprecated since Twisted Conch 0.9.  Use Key.fromString().
  'ssh-rsa' : keys.getPrivateKeyObject(data = privateKey)
```
This should really have been taken care of at the same time that ```getPrivateKeyObject``` was deprecated, but as long as we get to it before release it will be fine.

<details><summary>Searchable metadata</summary>

```
trac-id__2912 2912
type__defect defect
reporter__glyph glyph
priority__highest highest
milestone__ 
branch__branches_manhole_ssh_warning_2912 branches/manhole-ssh-warning-2912
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1195493588000000 1195493588000000
changetime__1196206723000000 1196206723000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

There are a lot of other deprecations coming out of Twisted from Conch APIs.  I'm not sure why manhole_ssh is singled out here.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

I didn't mean to single it out.  Obviously all the deprecations should be fixed :).  This particular deprecation shows up when running trial on just about any part of the Divmod code though, which is why I noticed it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Maybe in addition to fixing the usage of the deprecated API here, the plugin which imports this module at lookup time should be changed not to do that?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

FWIW, it isn't a plugin.  It's axiom.batch, importing manhole_ssh to allow debugging the batch process.  As far as I can tell, it isn't using any deprecated APIs itself, only non-deprecated functionality within Twisted whose implementation is still using deprecated APIs internally.  I don't think that updating it to not do this would be an improvement; although one could easily argue that there are better ways that debugging the batch process might be implemented, I don't think there's anything really wrong with using manhole_ssh (although I don't know if there's a better API to do what it's accomplishing).

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|pahan commented|
|-|-|

(In [21977]) Branching to 'manhole-ssh-warning-2912'

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|PenguinOfDoom set owner to @glyph|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set owner to PenguinOfDoom|
|-|-|

The change in transport.py is not great. You get the keyObject attribute, then rebuild a Key to make the verify: you can probably just keep the Key instance.

The rest is OK, it tries to keep the changes minimal, which I guess is cool because there are a few conch branches around. And the reported problem is fixed. So please merge when first point is addressed.

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|pahan set status to `closed`|
|-|-|

(In [21983]) Merge manhole-ssh-warning-2912

Author: pahan
Reviewer: therve
Fixes [#2912](https://github.com/twisted/twisted/issues/2912)

Fix t.conch.manhole_ssh and t.c.ssh.transport so that test_recvline passes without any warnings.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
